sinks/kafka/kafka.go

```diff
@@ -225,6 +229,19 @@ func NewKafkaSpanSink(logger *logrus.Logger, brokers string, topic string, parti
 		serializer = "protobuf"
 	}
 
+	var sampleThreshold uint32
+	if sampleRate <= 0.0 {
+		return nil, errors.New("Span sample rate must be greater than 0.0")
```

Comment:
Why not just default to 0 with a `Warn` since we do that in the `> 1.0` case?

Reply:
I don’t understand what we would want as a meaningful distinction between “sampleRate is 0.0” and “kafka sample sink is disabled”, so I did this. My gut feeling is that if someone sets a sample rate to `-1.0`, they don’t understand what the sample rate represents, and we should error – but honestly, the same logic _should_ be applied to > 1.0 in this case.

What do you think about me changing it from `sampleRate` to `sampleRatePercent` and making it an integer for now? That way we could bound it from `0` to `100` at the sacrifice of some granularity.

Reply:
If I imagine REALLY HARD I can think of someone wanting to turn it to 0 as a sort of feature flag. As in "o, let's disable this real fast" but that's just as easily done by not including the config keys.

Iow, I think you should probably leave it be! Sounds like more work that is useful to adjust it. The semantics you describe make sense.